tests/how_to_test_systems.rs

```diff
@@ -95,3 +138,40 @@ fn spawn_enemy_using_input_resource() {
     // Check resulting changes, no new entity has been spawned
     assert_eq!(world.query::<&Enemy>().iter(&world).len(), 1);
 }
+
+#[test]
+fn update_score_on_event() {
+    // Setup app
+    let mut app = App::new();
+
+    // Add Score resource
+    app.insert_resource(Score(0));
+
+    // Add `EnemyDied` event
+    app.add_event::<EnemyDied>();
+
+    // Add our systems
+    app.add_system(update_score);
+
+    // Send an `EnemyDied` event
+    app.world
+        .resource_mut::<Events<EnemyDied>>()
+        .send(EnemyDied(3));
+
+    // Run systems
+    app.update();
+
+    // Get `EnemyDied` event reader
+    let enemy_died_events = app.world.resource::<Events<EnemyDied>>();
+    let mut enemy_died_reader = enemy_died_events.get_reader();
+    let enemy_died = enemy_died_reader.iter(enemy_died_events).next().unwrap();
+
+    // Check the event has been sent
+    assert_eq!(enemy_died.0, 3);
```

Comment:
this is just asserting that the event you sent yourself is sent, it doesn't use anything from the system

Reply:
updated. was just trying to show how to read and write events in a test, but moved the read to `did_despawn_enemy`.